android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonFilePicker.kt

```diff
@@ -69,7 +74,11 @@ class AddonFilePicker(
         }
     }
 
-    internal fun convertToFileUri(uri: Uri): String = uri.toFileUri(context, "XPIs").toString()
+    internal fun removeTemporaryFiles() {
+        File(context.cacheDir, XPI_DIR).deleteRecursively()
```

Comment:
Why do we need the recursive behavior here? Can we use `delete()` instead?

Reply:
As we are deleting a directory `delete()` doesn't work.

Reply:
Wait, why are we deleting a directory here? Could we delete the file instead?

Reply:
If we delete the actual file, we won't be able deleting the previous stored files, side note: the directory is created every time we create a file if it's not required.

Reply:
I think I am fine with the trade-off, I'd rather delete a file than an entire directory given users can clear their cache to clear everything and this feature landed recently.

Reply:
👍🏽  I updated the patch!